functionary/ui/views/builds.py

```diff
@@ -0,0 +1,20 @@
+from builder.models import Build, BuildLog
+
+from .view_base import (
+    PermissionedEnvironmentDetailView,
+    PermissionedEnvironmentListView,
+)
+
+
+class BuildListView(PermissionedEnvironmentListView):
+    model = Build
+    order_by_fields = ["id"]
+
+
+class BuildDetailView(PermissionedEnvironmentDetailView):
+    model = Build
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["build_log"] = BuildLog.objects.filter(build=context["object"])[0]
```

Comment:
You can get the BuildLog through the Build object:

```python
context["build_log"] = context["object"].buildlog
```

Also, regardless of which way you do it, the build log will not exist prior to the build completing, so you'll have to handle that here.  One option is to catch ObjectDoesNotExist (from django.core.exceptions):

```python
try:
    context["build_log"] = context["object"].buildlog
except ObjectDoesNotExist:
    context["build_log"] = None